### PR TITLE
feat: Improve rg and highlight pattern matching

### PR DESCRIPTION
**Pull request by Mike325**
- Use \b to narrow down ripgrep results
- Highlight only keywords inside comments

### CODE REVIEW

**Comment by folke** [comment]
This looks great!  I won't have time till later today to test this.

File: README.md [hunk 1]
```diff
@@ -99,9 +99,10 @@ Todo comes with the following defaults:
   },
   -- regex that will be used to match keywords.
   -- don't replace the (KEYWORDS) placeholder
-  pattern = "(KEYWORDS):",
-  -- pattern = "(KEYWORDS)", -- match without the extra colon. You'll likely get false positives
-  -- pattern = "-- (KEYWORDS):", -- only match in lua comments
+  rg_pattern = '\\b(KEYWORDS):', -- rust regex
```
Comment (folke) [inline README.md:102]: rename docs to:
- ripgrep regex
- vim regex

File: lua/todo-comments/config.lua [hunk 2]
```diff
@@ -47,9 +47,10 @@ local defaults = {
   },
   -- regex that will be used to match keywords.
   -- don't replace the (KEYWORDS) placeholder
-  pattern = "(KEYWORDS):",
-  -- pattern = "(KEYWORDS)", -- match without the extra colon. You'll likely get false positives
-  -- pattern = "-- (KEYWORDS):", -- only match in lua comments
+  rg_pattern = '\\b(KEYWORDS):', -- rust regex
```
Comment (folke) [inline lua/todo-comments/config.lua:50]: rename docs to:

* ripgrep regex
* vim regex

File: plugin/todo.vim [hunk 3]
```diff
@@ -1,4 +1,6 @@
 
-command! TodoQuickFix lua require("todo-comments.search").setqflist()
-command! TodoTelescope Telescope todo-comments todo
-command! TodoTrouble Trouble todo
+if executable('rg')
```
Comment (folke) [inline plugin/todo.vim:2]: I'd rather not have this here.
When a user will try to execute `TodoTrouble` etc, they will get a proper error saying they need to have ripgrep installed

File: lua/todo-comments/highlight.lua [hunk 4]
```diff
@@ -34,9 +35,14 @@ function M.highlight(buf, first, last)
   end
 
   local lines = vim.api.nvim_buf_get_lines(buf, first, last + 1, false)
+  local comment_str = vim.bo.commentstring:gsub('%s+', '')
+  comment_str = vim.fn.escape(comment_str, '*+?')
+  local wrap = comment_str:sub(#comment_str - 1, #comment_str) ~= '%s'
+  local pattern = comment_str:gsub('%%s', [[\s*]]..(wrap and Config.hl_regex..'.*' or Config.hl_regex))
```
Comment (folke) [inline lua/todo-comments/highlight.lua:41]: Shouldn't this always be:
```lua
comment_str:gsub('%%s', [[\s*]].. Config.hl_regex ..'.*')
```
```

Comment (folke) [reply lua/todo-comments/highlight.lua:41]: I don't think the extra `wrap` check is needed?

**Review by folke — CHANGES_REQUESTED** [review]
Apart from my inline comments, I would also rather have the comment matching behind a config option `highlight_comments_only`.

Thank you!

File: lua/todo-comments/highlight.lua (hunk 4 above)
Comment (Mike325) [reply lua/todo-comments/highlight.lua:41]: yep, you're right, .* should work

**Review by Mike325 — COMMENTED** [review]
> 
> 
> Apart from my inline comments, I would also rather have the comment matching behind a config option `highlight_comments_only`.
> 
> Thank you!

It may be better if all highlight options into the highlight property
```lua
highlight = {                                                                               
  before = "",
  keyword = "wide",
  after = "fg",
  comments_only = true,
  pattern = '<(KEYWORDS)>:', 
},
```

**Comment by folke** [comment]
Good point. Then probably also add the other one like this:

```lua
search = {
    pattern = "...", -- ripgrep search pattern
}
```

I'll add in the `rg` default args there later so definitely useful to have it as a group under options too

**Comment by folke** [comment]
Thanks again for this PR. I've merged it in and made some additional changes to the default patterns and make sure it never uses the comment string for the quickfix buffers

**Pull request merged**